atomsci/ddm/test/unit/test_perf_data.py

```diff
@@ -0,0 +1,339 @@
+    # This should have r2 of 1
+    r2 = perf.accumulate_preds(pred_vals, ids)
+    assert r2 == 1
+    # do a few more folds
+    r2 = perf.accumulate_preds(pred_vals, ids)
+    r2 = perf.accumulate_preds(pred_vals, ids)
+
+    (res_ids, res_vals, res_std) = perf.get_pred_values()
+    (r2_mean, r2_std) = perf.compute_perf_metrics()
+
+    assert np.allclose(res_vals, real_vals)
+    assert np.allclose(res_std, np.zeros_like(res_std))
+
+    # perfect score every time
+    assert r2_mean==1
+    assert r2_std==0
+
+def test_KFoldClassificationPerfData():
+    res_dir, tmp_dskey = setup_paths()
+
+    params = {"verbose": "True",
+    "datastore": "False", 
+    "save_results": "False", 
+    "model_type": "NN", 
+    "featurizer": "ecfp", 
+    "split_strategy": "k_fold_cv", 
+    "splitter": "random",
+    "split_test_frac": "0.15", 
+    "split_valid_frac": "0.15", 
+    "transformers": "True", 
+    "dataset_key": tmp_dskey,
+    "id_col": "compound_id", 
+    "response_cols":"active", 
+    "smiles_col": "base_rdkit_smiles",
+    "max_epochs":"2",
+    "prediction_type": "classification",
+    "result_dir":res_dir}
+
+    # setup a pipeline that will be used to create performance data
+    pparams = parse.wrapper(params)
+    mp = model_pipeline.ModelPipeline(pparams)
+    mp.train_model()
+
+    # creat performance data
+    perf = perf_data.create_perf_data(mp.params.prediction_type, 
+            mp.data, mp.model_wrapper.transformers, 'train')
+
+    assert isinstance(perf, perf_data.KFoldClassificationPerfData)
+
+    ids = sorted(list(mp.data.train_valid_dsets[0][0].ids))
+    weights = perf.get_weights(ids)
+    assert weights.shape == (len(ids),1)
+    assert all(weights==1)
+
+    real_vals = perf.get_real_values(ids)
+    d = dc.data.NumpyDataset(X=np.ones_like(real_vals), y=real_vals, ids=ids, w=np.ones(len(ids)))
+    # There should be no transformers
+    assert len(perf.transformers) == 0
+
+    num_classes = 2
+    # input to to_one_hot needs to have the shape (N,) not (N,1)
+    pred_vals = dc.metrics.to_one_hot(d.y.reshape(len(d.y)), num_classes)
+    # This should have r2 of 1
+    roc_auc_score = perf.accumulate_preds(pred_vals, ids)
+    assert roc_auc_score == 1
+    # do a few more folds
+    roc_auc_score = perf.accumulate_preds(pred_vals, ids)
+    roc_auc_score = perf.accumulate_preds(pred_vals, ids)
+
+    (res_ids, res_classes, res_probs, res_std) = perf.get_pred_values()
+    (roc_auc_mean, roc_auc_std) = perf.compute_perf_metrics()
+
+    # std should be zero
+    assert all((res_std==np.zeros_like(res_std)).flatten())
+    # probs should match predictions
+    assert all((res_probs==pred_vals.reshape(len(d.y), 1, num_classes)).flatten())
+    # all predictions are correct
+    assert all(res_classes==real_vals)
+    # perfect score every time
+    assert roc_auc_mean==1
+    assert roc_auc_std==0
+
+def test_SimpleRegressionPerfData():
+    res_dir, tmp_dskey = setup_paths()
+
+    params = {"verbose": "True",
```

Comment:
seems like the convention is now to create a json file in a separate folder to encapsulate this info

Reply:
I made a json file, but kept it in the same folder since that's what the other unit tests did.